app/src/main/java/com/sci/recipeandroid/feature/auth/ui/viewmodel/AuthOptionViewModel.kt

```diff
@@ -0,0 +1,89 @@
+package com.sci.recipeandroid.feature.auth.ui.viewmodel
+
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.credentials.exceptions.GetCredentialCancellationException
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sci.recipeandroid.feature.auth.data.service.FacebookAuthenticator
+import com.sci.recipeandroid.feature.auth.data.service.GoogleAuthenticator
+import com.sci.recipeandroid.feature.auth.domain.repository.AuthRepository
+import com.sci.recipeandroid.util.SingleLiveEvent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class AuthOptionViewModel(
+    private val authRepository: AuthRepository,
+    private val googleAuthenticator: GoogleAuthenticator,
+    private val facebookAuthenticator: FacebookAuthenticator,
+) : ViewModel() {
+    private val _uiEvent = SingleLiveEvent<String>()
+    val uiEvent: LiveData<String> = _uiEvent
+
+    fun googleAuthentication(context: Context) {
```

Comment:
method name

Reply:
changed this function name at 9baf80468f57d78d6c4b13f545e1ebaa10f9d59c